RecoBTag/DeepFlavour/interface/c_pf_features_converter.h

```diff
@@ -0,0 +1,263 @@
+                               const float & drminpfcandsv,
+                               ChargedCandidateFeaturesType & c_pf_features) {
+
+            // track_info has to be built before passing as parameter
+
+            // could be inlined
+            float etasign = 1.;
+            if (jet.eta()<0) etasign =-1.;
+
+            c_pf_features.pt = c_pf->pt();
+            c_pf_features.eta = c_pf->eta();
+            c_pf_features.phi = c_pf->phi();
+            c_pf_features.ptrel = deep::catch_infs_and_bound(c_pf->pt()/jet.pt(),
+                                                             0,-1,0,-1);
+            c_pf_features.erel = deep::catch_infs_and_bound(c_pf->energy()/jet.energy(),
+                                                            0,-1,0,-1);
+            c_pf_features.phirel = deep::catch_infs_and_bound(fabs(reco::deltaPhi(c_pf->phi(),jet.phi())),
+                                                              0,-2,0,-0.5);
+            c_pf_features.etarel = deep::catch_infs_and_bound(fabs(c_pf->eta()-jet.eta()),
+                                                              0,-2,0,-0.5);
+            c_pf_features.deltaR = deep::catch_infs_and_bound(reco::deltaR(*c_pf,jet),
+                                                             0,-0.6,0,-0.6);
+            c_pf_features.dxy = deep::catch_infs_and_bound(fabs(c_pf->dxy()),0,-50,50);
+
+
+            c_pf_features.dxyerrinv = deep::catch_infs_and_bound(1/c_pf->dxyError(),
+                                                                 0,-1, 10000.);
+
+            c_pf_features.dxysig = deep::catch_infs_and_bound(fabs(c_pf->dxy()/c_pf->dxyError()),
+                                                              0.,-2000,2000);
+
+
+            c_pf_features.dz = c_pf->dz();
+            c_pf_features.VTX_ass = c_pf->pvAssociationQuality();
+            c_pf_features.fromPV = c_pf->fromPV();
+            c_pf_features.vertexChi2=c_pf->vertexChi2();
+            c_pf_features.vertexNdof=c_pf->vertexNdof();
+            c_pf_features.vertexNormalizedChi2=c_pf->vertexNormalizedChi2();
+            c_pf_features.vertex_rho=deep::catch_infs_and_bound(c_pf->vertex().rho(),
+                                                                0,-1,50);
+            c_pf_features.vertex_phirel=reco::deltaPhi(c_pf->vertex().phi(),jet.phi());
+            c_pf_features.vertex_etarel=etasign*(c_pf->vertex().eta()-jet.eta());
+            c_pf_features.vertexRef_mass=c_pf->vertexRef()->p4().M();
+
+            c_pf_features.puppiw = c_pf->puppiWeight();
+
+            const auto & pseudo_track =  c_pf->pseudoTrack();
+            auto cov_matrix = pseudo_track.covariance();
+
+            c_pf_features.dptdpt =    deep::catch_infs_and_bound(cov_matrix[0][0],0,-1,1);
+            c_pf_features.detadeta=   deep::catch_infs_and_bound(cov_matrix[1][1],0,-1,0.01);
+            c_pf_features.dphidphi=   deep::catch_infs_and_bound(cov_matrix[2][2],0,-1,0.1);
```

Comment:
do the negative diagonal values of cov make sense as inputs in this context? a lower bound of 0 appears more meaningful .. in case cov comes out corrupt with some negative values

Reply:
@slave what your are saying makes sense, only concern is consistency with traning. @mverzett @jkiesele, what do you think about this?